src/PageHeader/manageResponsiveValueWithCSS.ts

```diff
@@ -0,0 +1,53 @@
+import {isResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {BetterSystemStyleObject} from '../sx'
+
+/**
+ * This function is inspired by the `useResponsiveValue` hook and it's used to render responsive values with CSS.
+ * @param value - The value that we want to render responsively
+ * @param fallback - The fallback value
+ * @param cssProperty - The CSS property that we want to render
+ * @param truety - The value that we want to render when the value is true
+ * @param falsy - The value that we want to render when the value is false
+ 
+ * @example
+ * CSSManagedResponsiveValue({narrow: true, regular: true, wide: false}, false, 'display', 'none', 'flex')
+ * @returns
+ * {
+ *   "@media screen and (max-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "display": "flex"
+ *   }
+ * }
+ */
+export function CSSManagedResponsiveValue<T, F, B, C>(
+  value: T,
+  fallback: F,
+  cssProperty: string,
+  truety: B,
+  falsy: C,
+): BetterSystemStyleObject {
+  if (isResponsiveValue(value)) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const responsiveValue = value as Extract<T, ResponsiveValue<any>>
+    const checkFallback = fallback ? truety : falsy
+
+    return {
```

Comment:
One thing that seems tricky about this is that it's always generating media queries even if we may not need them. For example,

```js
const hiddenOnRegularAndWide = {
  narrow: false,
  regular: true,
  wide: true,
}
```

For this object, we would only need two breakpoints, `narrow` and `regular`, since wide would already be covered by `regular`.

Curious if there would be any way to simplify this, update the default values, or if it's not worth going through that process at the moment

Reply:
This is a great feedback @joshblack! Thank you so much. I addressed this in my new implementation, would love to hear what you think 🙌🏼